<div class="alert alert-block alert-danger">

# Addition Rule (COMPLETE)

</div>

Made in collaboration with [Skew the Script](https://skewthescript.org/) and [CourseKata](https://coursekata.org/).

<img src="https://i.postimg.cc/ty1GkxB8/Skew-the-Script-Logo.png" title="Skew the script logo" width=200 align = left>

<img src="https://i.postimg.cc/tXcF0nzD/Course-Kata-logo.png" title="CourseKata logo" width=200 align = right>

In [ ]:
# Load the CourseKata library
suppressPackageStartupMessages({
    library(coursekata)
})

## 1.0 - Online Dating Profiles

Have you or anyone you know ever created an online dating profile? What was the experience like? Do you think most people are completely honest when creating an online profile?

**Well, let's consider the following scenario:**

<img src="https://i.postimg.cc/tXDghGz0/Addition-Rule-Man-Wonders2.png" title="A Man Wonders" alt= "a 36-year-old heterosexual man wonders to himself: How can I make my online profile more appealing?" width= 500/>

 

**1.1:  What sort of messages will pop culture (such as romantic comedies) suggest to him about what is most desirable?**



<img src="https://i.postimg.cc/Bszh7Xgm/Addition-Rule-Media-Influence.png" title="Romantic Movie Posters" width = 800/>

 

<div class="alert alert-block alert-warning">

**Sample Responses**

Pop culture may suggest that our insidious social standards favor:

- Tall
- High income
- White 

</div>

So, perhaps this man thinks to himself....*"To be more appealing, maybe I should make myself appear a little taller, and a little wealthier in my online profile."*



<img src="https://i.postimg.cc/D21ZmbSz/Addition-Rule-Tall-Wealthy2.png" title="Tall Wealthy Man" width = 500/>

 

#### Key Analysis

Let's use some data to answer the question: 

**What percent of heterosexual men are insecure enough to lie online?**

To answer this, we will need to know the probability of randomly selecting an online profile of a tall and wealthy man, so we will need to develop a probability model.

To help us analyze this, let's first get familiar with some useful notation.

### 2.0 - Ways to Express Probability

**Probability Notation**

Since we are trying to figure out the probability of some event, we will need some ways to put our words into math language. 

For instance, we can say *"the probabilty of Event X"* like this: 

$P(X)$

In this case, we may want to figure out *"the probability of being tall"*, so, in math terms, we could say it this way: 

$P(Tall)$

**2.1:** How might we express *"The probability of being wealthy"*?

<div class="alert alert-block alert-warning">

**Sample Response**

$P(Wealthy)$

</div>

We can also expand that notation to make statements about the probability of being tall AND wealthy, or being tall OR wealthy.

These are mathematical ways to say these words:

- $\cup$ is the symbol for **Union** and is a way to say **"Or"**
- $\cap$ is the symbol for **Intersection** and is a way to say **"And"**

**2.2:** For example, which of these says *"the probability of being tall and wealthy"*? What does the other expression say?

> **A.** $P(Tall \cap Wealthy)$

> **B.** $P(Tall \cup Wealthy)$

<div class="alert alert-block alert-warning">

**Sample Response**

The probability of being tall AND wealthy:

> **A.** $P(Tall \cap Wealthy)$

The probability of being tall OR wealthy:

> **B.** $P(Tall \cup Wealthy)$

</div>

**Probabilities Visualized**

We can visualize these ideas with Venn Diagrams as well! Let's imagine we have two possible events: **Event A** (e.g., being Tall) and **Event B** (e.g., being Wealthy), and the part where they overlap is their intersection (e.g., being Tall and Wealthy). 

We can represent this like so:

<img src="https://i.postimg.cc/LHW4zN39/Addition-Rule-Two-Events.png" title="Two Events: A&B" width = 200/>

**2.3:** Looking at the way the diagrams are shaded below, match the following expressions to the diagram that represents that possibility.
 
1. $P(A)$
2. $P(B)$ 
3. $P(A \cup B)$
4. $P(A \cap B)$

<img src="https://i.postimg.cc/GLwgBz1h/Addition-Rule-Venn-Diagrams.png" title="Venn Diagrams" width = 600/>


<div class="alert alert-block alert-warning">

**Sample Response**

1. $P(A)$ --> Diagram B
2. $P(B)$ --> Diagram D
3. $P(A \cup B)$ --> Diagram C
4. $P(A \cap B)$ --> Diagram A

</div>

### 3.0 - The Data

Now, let's use these ideas with our data analysis.

The `profiles` dataset below is composed of public online dating profiles from 2012. We filtered for people who identified as age 36, male, and heterosexual, living in the San Francisco area. After filtering, we collected data for each profile that reported their height and yearly earnings.

***Why are we looking at just the age 36 profiles?***

- It’s a lower-bound for the age when men typically earn their median salaries.
- It’s an age when men who are dating online might really start feeling the pressure to find someone.

**3.1:** Take a look at the data frame. What does each case represent?

In [ ]:
profiles <- read.csv("https://docs.google.com/spreadsheets/d/e/2PACX-1vT-uzkNU7wCiNtwmk4mehDYGgz1bmYP-w6Uoo3x5N5WCOY2lceCueKW4RVHnGr0PMI9p4eCaWbCvaDI/pub?gid=950707983&single=true&output=csv")
head(profiles)

username        age height income gender orientation
1 user_0000035508 36  67      80000 m      straight   
2 user_0000035888 36  67      60000 m      straight   
3 user_0000035928 36  67      20000 m      straight   
4 user_0000035940 36  62      30000 m      straight   
5 user_0000034031 36  73     150000 m      straight   
6 user_0000036283 36  74     100000 m      straight

<div class="alert alert-block alert-warning">

**Sample Response**

The cases are individual public dating profiles.

</div>

**Median Height and Median Income**

- The estimated 2012 median yearly earnings for individual men in San Francisco county: **\$59,397** 

- Male median height (U.S.): **69.2 in. (5’9”)**

*Source Note: Income median is from the 2012 American Community Survey (data.census.gov) and height median is from the Center for Disease Control and Prevention, 2018.*

**3.2:** The median is the 50th percentile. This means we would expect what percent of people to be above it or below it?

<div class="alert alert-block alert-warning">

**Sample Response**

We would expect 50% of the people to be above it and and 50% to be below it.

</div>

Taking all of this into account, we can categorize the 192 men in the `profiles` dataset using the following convention:

<img src="https://i.postimg.cc/DvSm1Thp/Addition-Rule-Table-of-Categories.png" title="Table of Categories" width = 800/>

**3.3:** What should we expect the probabilities for each possible outcome to be if they are being generated by a process where everyone is reporting the information in their profiles honestly? For instance, what is the probability of randomly selecting the profile of a Tall Low-Earner?

Let: 

Tall = event of selecting a tall man

Wealthy = event of selecting a high-income earner

Under the assumption that people are reporting honestly, indicate the expected probabilities:

1. $P(Tall)$
2. $P(Wealthy)$
3. $P(Tall \cup Wealthy)$
4. $P(Tall \cap Wealthy)$

<div class="alert alert-block alert-warning">

**Sample Response**

Under the assumption that people are reporting honestly:

1. $P(Tall) = .50$
2. $P(Wealthy) = .50$
3. $P(Tall \cup Wealthy) = .75$
4. $P(Tall \cap Wealthy) = .25$

</div>

**3.4:** Does the assumption that the DGP comes from "honest" reports reflect the empty model or an alternative model? Why?

<div class="alert alert-block alert-warning">

**Sample Responses**

- It reflects the empty model, because that model assumes no differences among the groups. 

- It takes the position that there is no effect of lying or exaggerating on profiles, and that they should roughly match the patterns seen in the larger population.

</div>

### 4.0 - What is being reported in these profiles?

**4.1:** We will need to write some code to create height and income variables that are split at the median (the medians of the broader population listed in the previous section, not the medians of our sample). Save them back into your data frame as `height2` and `income2`. The code for `height2` has already been done for you.

In [ ]:
# Create a new variable by splitting 'height' at the median height of US men
profiles$height2 <- profiles$height <= 69.2
profiles$height2 <- factor(profiles$height2, levels = c(TRUE,FALSE), labels = c("Short", "Tall"))

# Create a new variable that splits income at the median for San Francisco
# COMPLETE VERSION
profiles$income2 <- profiles$income <=59397
profiles$income2 <- factor(profiles$income2, levels = c(TRUE, FALSE), labels = c("Low-Earner", "High-Earner"))

head(profiles)

username        age height income gender orientation height2 income2    
1 user_0000035508 36  67      80000 m      straight    Short   High-Earner
2 user_0000035888 36  67      60000 m      straight    Short   High-Earner
3 user_0000035928 36  67      20000 m      straight    Short   Low-Earner 
4 user_0000035940 36  62      30000 m      straight    Short   Low-Earner 
5 user_0000034031 36  73     150000 m      straight    Tall    High-Earner
6 user_0000036283 36  74     100000 m      straight    Tall    High-Earner

**4.2:** Next, Take a look at the counts for each `height2` and `income2` category combination in the table below. What do you notice?

In [ ]:
tally(height2 ~ income2, data = profiles, margins = TRUE)

income2
height2 Low-Earner High-Earner
  Short 21          48        
  Tall  22         101        
  Total 43         149

<div class="alert alert-block alert-warning">

**Sample Responses**

- There is not an even split between Low- and High-Earners

- There are many more High-Earners

- The category with the highest count is the Tall/High-Earner category

</div>

#### Addition Rule in Probability Models

**4.3:** We can turn these counts into proportions to create a probability model. Let's divide each group count by the total number of cases.


In [ ]:
"Short/Low-Earner"
21/192
"Short/High-Earner"
48/192
"Tall/Low-Earner"
22/192
"Tall/High-Earner"
101/192

[1] "Short/Low-Earner"

[1] 0.109375

[1] "Short/High-Earner"

[1] 0.25

[1] "Tall/Low-Earner"

[1] 0.1145833

[1] "Tall/High-Earner"

[1] 0.5260417

> We have generated the following probability model based on our sample:

<img src="https://i.postimg.cc/zDVXk8d1/Addition-Rule-P-Model.png" title="Probability Model" width = 800/>

We should then do the following validity check, to make sure our numbers make sense:

> 1. All probabilities are between 0 – 1 (true)
> 2. All probabilities add to 1 (true)

Now, we can use this probability model to figure out the probability of various outcomes such as: 

***"What is the probability of observing a profile that reports being taller than the median"*** 

We can simply **add the proportions** for any "Tall" categories, like so:

<img src="https://i.postimg.cc/BJdYSpW5/Addition-Rule-P-of-Tall.png" title="Probability of Being Tall" width = 800/>

**4.4:** We can see that we get a probability of 64.1%, which is a higher probability than in the whole population of U.S. men (50%)! Does this mean that 14.1% of these men are lying about their height on their profiles?

<div class="alert alert-block alert-warning">

**Sample Responses**

- Students may say yes, or still be unsure.

They may say things such as:

- Maybe they were just making a mistake

- Maybe the people making the profiles really are all the taller and wealthier people in San Francisco

</div>

**4.5:** We can also use R to take some shortcuts in figuring out various probabilities. Try modifying the code below to estimate *the probability of finding someone who is wealthier than the median*. Is it close to 50%? Much higher? Lower? What does this tell us?

In [ ]:
# We can use tally() to get the number of profiles that count as "Tall"
tally(~ height2=="Tall", data = profiles)

# We can even divide the tally() by the total number of profiles to get the proportion--the P(T)
tally(~ height2=="Tall", data = profiles) / 192

# Copy and paste the code above and try modifying it to estimate P(H)

# COMPLETE VERSION
tally(~ income2=="High-Earner", data = profiles)
tally(~ income2=="High-Earner", data = profiles) / 192

height2 == "Tall"
 TRUE FALSE 
  123    69 

height2 == "Tall"
    TRUE    FALSE 
0.640625 0.359375 

income2 == "High-Earner"
 TRUE FALSE 
  149    43 

income2 == "High-Earner"
     TRUE     FALSE 
0.7760417 0.2239583 

<div class="alert alert-block alert-warning">

**Sample Response**

The probability of someone wealthier than the median is almost 78%.

This tells us that potential dishonesty about wealth among profiles occurs at even higher rates than for height.

</div>

### 5.0 - Estimating $P(T \cup H)$ and $P(T \cap H)$ 

**INTERSECTION ($\cap$): P(T "AND" H)**

If our goal is to estimate the probability of finding someone who is taller AND wealthier than the median--**P(T $\cap$ H)**--it can be helpful to think of the $\cap$ to mean **"and"** or **"both"**. This means they have to be part of *both* the tall *and* the wealthy group (the intersection of the two categories). 

We can visualize this idea in a Venn diagram or a table. 

If we think of this as the entire event space (all observed outcomes):

<img src="https://i.postimg.cc/6Bh3qDPB/Addition-Rule-Event-Space.png" title="Event Space" width = 700/>

Then, we can count the observations for the space (or spaces) we are trying to estimate. In this case, we are interested in the intersection of the "Tall" and "High-Earner" groups. There is only one space that meets that condition:



<img src="https://i.postimg.cc/kCf2x1mT/Addition-Rule-P-of-T-and-H-Diagrams.png" title="Probability of T and H" width = 600/>

In [ ]:
# Find the probability by dividing by N
101 / 192

[1] 0.5260417

We can also use the `&` symbol in R to help us estimate this intersection!

In [ ]:
# Intersection = &
tally(~ (height2=="Tall" & income2=="High-Earner"), data = profiles)

# Divide by N to get the probability
tally(~ (height2=="Tall" & income2=="High-Earner"), data = profiles) / 192

(height2 == "Tall" & income2 == "High-Earner")
 TRUE FALSE 
  101    91 

(height2 == "Tall" & income2 == "High-Earner")
     TRUE     FALSE 
0.5260417 0.4739583 

**5.1:** Is this probability close to what we would expect if all the profile creators were reporting their information accurately?

<div class="alert alert-block alert-warning">

**Sample Response**

No, this probability (about 53%) is much higher than we would expect under the null model (which predicts 25%).

</div>

**UNION ($\cup$) = P(T "OR" H)** 

If we want to estimate the probability of finding someone who is tall, OR wealthy, OR both tall and wealthy, then that means we ***add*** all those possibilities together.

**Take Note:**

- $\cup$ = "OR" = "one, the other, or both"
- $\cup$ = "add" = P(T) + P(H) - P(T$\cap$ H)

In other words, below we can see that P(T) = 22 + 101, and P(H) = 48 + 101. Since we don't want to double count the intersection (101), we would just subtract that, or be careful not to double count it!

<img src="https://i.postimg.cc/X4cSChMJ/Addition-Rule-P-of-T-or-H-Diagrams.png" title="Probability of T or H" width = 600/>

In [ ]:
# Add the counts then divide by N to get the probability
(22 + 101 + 48) / 192

[1] 0.890625

We can also use the `|` symbol in R to help us estimate this union!

*NOTE: it is the vertical line symbol--not a lower case "L"--and is usually paired with the "backslash" (\\) key near the "backspace" or "delete" key on your keyboard.*

In [ ]:
# Union = |
tally(~ (height2=="Tall" | income2=="High-Earner"), data = profiles)

# Divide by N to get it as a proportion
tally(~ (height2=="Tall" | income2=="High-Earner"), data = profiles) / 192

(height2 == "Tall" | income2 == "High-Earner")
 TRUE FALSE 
  171    21 

(height2 == "Tall" | income2 == "High-Earner")
    TRUE    FALSE 
0.890625 0.109375 

**5.2:** What is the difference between what is expected and what we observe? Is that a lot?

<div class="alert alert-block alert-warning">

**Sample Response**

The difference between expected (75%) and observed(89%) is 14%.

That seems like a pretty big difference.

</div>

### 6.0 - A DGP of honesty?

**6.1:** Summarize the probabilities we found today in our sample:

1. $P(T)$ = 
2. $P(H)$ = 
3. $P(T \cup H)$ = 
4. $P(T \cap H)$ = 

Compare them to our expected probabilities for a DGP of "honest" profiles:

1. $P(T)$ = .50
2. $P(H)$ = .50
3. $P(T \cup H)$ = .75
4. $P(T \cap H)$ = .25

<div class="alert alert-block alert-warning">

**Sample Response**

Probability Model based on our sample:

1. $P(T)$ = 64.1%
2. $P(H)$ = 77.6%
3. $P(T \cup H)$ = 89.1%
4. $P(T \cap H)$ = 52.6%

</div>

**6.2:** Looking at the intersections, is there a group that seems to be the "most honest"?

<div class="alert alert-block alert-warning">

**Sample Response**

The Short High-Earners have the smallest discrepancy between what is expected and what is observed.

</div>

**6.3:** What are your overall conclusions? Are 36-year-old, heterosexual males typically being honest when reporting their height and income on thier online dating profiles? 

<div class="alert alert-block alert-warning">

**Sample Response**

Although not conclusive, there is fairly strong evidence that many of the men on OKCupid are lying. 64.1% reported they are above the median height and 77.6% reported they earn above the median income (among 36-year-old men in SF). However, those proportions are only 50% in the matching population of San Francisco.

Furthermore, in the United States, median income tends to rise slowly for men, nationally peaking at age 55. By contrast, in the OKCupid data, incomes spiked until age 32, leveling out after that. This suggests that men on the site feel increasing pressure to inflate their earnings as they approach their 30s, before hitting a “lying threshold,” where they won’t inflate their incomes further after age 32 (if you lie too much, the lie becomes unreasonable). 

There is a chance that men who choose to make profiles on the site truly tend to be taller and wealthier than the general population. However, previous studies have found single men tend to be less wealthy than their married peers (see [article](https://research.stlouisfed.org/publications/economic-synopses/2018/09/14/married-men-sit-atop-the-wage-ladder)).

</div>